param/parameterized.py

```diff
@@ -913,6 +926,44 @@ def self_or_cls(self_):
         return self_.cls if self_.self is None else self_.self
 
 
+    def __getitem__(self_, key):
+        inst = self_.self
+        parameters = self_.params() if inst is None else inst.param.params()
+        p = parameters[key]
+        if inst is not None and p.per_instance and key not in inst._instance__params:
+            p = copy.copy(p)
+            p._owner = inst
+            inst._instance__params[key] = p
+        return p
+
+
+    def __dir__(self_):
```

Comment:
Last I checked IPython isn't respecting ``__dir__` properly if you want this for tab completion purposes...

Reply:
Works for me.

Reply:
Probably works for some IPython versions and not others.